fip-0039.md

```diff
@@ -0,0 +1,155 @@
+
+An improvement has been suggested to have a separate key for receiving tokens and for encryption. This would allow for a different security scheme to be applied to each key. In fact, the account owner may not even have access to a private key designated as the receiving key for tokens, e.g. it’s in cold storage, while they do need to have access to the private key in order to decrypt some FIO Chain data.
+
+It is important to note, that the encryption key is only used by the client and has no use on-chain. Here's how is encryption key used:
+
+#### Sender is encrypting data intended for receiver
+
+|Step|Before FIP|After FIP|
+|---|---|---|
+|Fetch receiver's FIO Public Key to use for encryption.|_get_pub_address_ for sender's _fio_address_, _chain_code: FIO_, _token_code: FIO_|_get_encrypt_key_ for sender's _fio_address_|
+|Encrypt the data|Client process|No change|
+|Send on-chain transaction|_newfundsreq_ or _recordobt_|No change|
+
+#### Receiver is decrypting the data from Sender
+
+|Step|Before FIP|After FIP|
+|---|---|---|
+|Read FIO Public Key to use for decryption from the specific record|_get_received_fio_requests_, _get_sent_fio_requests, _get_pending_fio_requests_, _get_obt_data_, _get_cancelled_fio_requests_|No change|
+|Decrypt the data|Client process|No change|
+
+# Specification
+## New actions
+### Update encryption key
+Updates the FIO Public Key to use for encryption of New Funds Request/Record OBT content blob.
+#### Contract: fio.address
+#### New action: *updcryptkey*
+#### New end point: /update_encrypt_key
+#### New fee: update_encrypt_key, bundle-eligible (1 bundled transaction) fee amount will be determined during development and updated here
+#### RAM increase: To be determined during implementation
+#### Request body
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_address|Yes|String|FIO Address to add encrypto key for.|
+|encrypt_public_key|Yes|Valid FIO Public Key|FIO Public key to use for encryption.|
+|max_fee|Yes|Positive Int|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by [/get_fee](https://developers.fioprotocol.io/api/api-spec/reference/get-fee/get-fee) for correct value.|
+|actor|Yes|12 character string|Valid actor of signer|
+|tpid|Yes|FIO Address|FIO Address of the entity which generates this transaction. TPID rewards will be paid to this address. Set to empty if not known.|
+##### Example
+```
+{
+  "fio_address": "purse@alice",
+  "encrypt_public_key": "FIO5G2tfhPZtiaiBU6o9SpT2XKbb28gtKUeXXarvBjxuU29VYgYCs",
+  "max_fee": 0,
+  "actor": "aftyershcu22",
+  "tpid": "rewards@wallet"
+}
+```
+#### Processing
+* Request is validated per Exception handling.
+  * Explicitly allowed
+    * Setting of encrypt_public_key to empty string
+* Supplied encryption key is added to FIO Address.
```

Comment:
Adding new fields to tables that contain data requires a data migration (casey to confirm).